boards: shields: Fix adafruit_2_8_tft_touch_v2 Arduino D4 pin (SDMMC CS)

MaureenHelm opened the pull request:
The device tree overlay for the adafruit_2_8_tft_touch_v2 shield board
was incorrectly mapping the Arduino D4 pin, used for SDMMC chip select.

Signed-off-by: Maureen Helm <maureen.helm@nxp.com>

MaureenHelm committed:
boards: shields: Fix adafruit_2_8_tft_touch_v2 Arduino D4 pin (SDMMC CS)

The device tree overlay for the adafruit_2_8_tft_touch_v2 shield board
was incorrectly mapping the Arduino D4 pin, used for SDMMC chip select.

Signed-off-by: Maureen Helm <maureen.helm@nxp.com>

MaureenHelm committed:
boards: shields: Enable SDMMC disk driver on adafruit_2_8_tft_touch_v2

Configures the adafruit_2_8_tft_touch_v2 shield board to enable SDMMC by
default when disk drivers are enabled.

Signed-off-by: Maureen Helm <maureen.helm@nxp.com>

MaureenHelm committed:
boards: arm: Configure frdm_k64f Arduino D4 pin as gpio

Configures the pinmux for the Arduino D4 pin to allow using it as a
gpio. This is needed by the adafruit_2_8_tft_touch_v2 shield for SDMMC.

Signed-off-by: Maureen Helm <maureen.helm@nxp.com>

MaureenHelm committed:
samples: fat_fs: Build with adafruit_2_8_tft_touch_v2 shield

Adds a new sample configuration to build the fat_fs sample with the
adafruit_2_8_tft_touch_v2 shield and SDMMC SPI driver in CI. It has been
tested with frdm_k64f and mimxrt685_evk_cm33 boards, but should work
with any board that supports Arduino SPI and Arduino GPIO features.

Signed-off-by: Maureen Helm <maureen.helm@nxp.com>

jfischer-no commented on boards/shields/adafruit_2_8_tft_touch_v2/adafruit_2_8_tft_touch_v2.overlay:
```diff
@@ -9,7 +9,7 @@
 &arduino_spi {
 	status = "okay";
 	cs-gpios = <&arduino_header 16 GPIO_ACTIVE_LOW>,	/* D10 */
-		   <&arduino_header 12 GPIO_ACTIVE_LOW>;	/* D04 */
```
I messed it up, for the [epaper shield](https://github.com/zephyrproject-rtos/zephyr/blob/master/boards/shields/waveshare_epaper/dts/waveshare_epaper_common.dtsi#L10) the comment is wrong but gpio pin is correct.

erwango commented on samples/subsys/fs/fat_fs/sample.yaml:
```diff
@@ -10,3 +10,7 @@ tests:
   sample.filesystem.fat_fs.overlay:
     platform_allow: nrf52840_blip
+  sample.filesystem.fat_fs.adafruit_2_8_tft_touch_v2:
+    platform_allow: mimxrt685_evk_cm33 frdm_k64f
```
What about `    depends_on: arduino_i2c arduino_gpio` ?

MaureenHelm replied:
Good idea, done.

MaureenHelm committed:
boards: shields: Enable gpio by default on adafruit_2_8_tft_touch_v2

The display controller and SDMMC slot both define GPIOs for their chip
select signals, so enable GPIO drivers by default for this shield.

This missing default was discovered in CI when extending the fat_fs
sample to build with this shield. Many boards enable GPIO drivers by
default, but some boards like sam_v71_xult and sam_v71b_xult do not.

Signed-off-by: Maureen Helm <maureen.helm@nxp.com>
